internal/net/tcp/dialer.go

```diff
@@ -20,15 +20,23 @@ package tcp
 import (
 	"context"
 	"crypto/tls"
-	"strings"
+	"fmt"
+	"math"
+	"sync/atomic"
 	"time"
 
 	"github.com/vdaas/vald/internal/cache"
+	"github.com/vdaas/vald/internal/errors"
 	"github.com/vdaas/vald/internal/log"
 	"github.com/vdaas/vald/internal/net"
 	"github.com/vdaas/vald/internal/safety"
 )
 
+const (
+	maxDialerCacheCount = math.MaxUint32 - 10000
```

Comment:
Could you please teach me about `- 10000` ?

Reply:
it is just a buffer. When there are multiple threads calling `GetIP()` function, there is a chance that adding the `cnt` to overflow, so instead of define it as `math.MaxUint32`, I think -10000 is more the enough for the buffer.